build.gradle

```diff
@@ -54,6 +55,7 @@ ext {
 // if the build file is incorrectly written and evaluates something
 // eagerly).
 
+apply from: file('gradle/wrapper.gradle')
```

Comment:
We don't need to set up the wrapper task. If we can't have the local jar then the launch scripts must take care to download the proper version prior to launching gradle.

Reply:
We don't strictly need this, but the version to download ends up hard-coded in the wrapper, so upgrading gradle in the future would need us to regenerate. Not a great option, but maybe not the worst. Can remove this.

Reply:
Gradle version to download should be in gradle-wrapper.properties. The wrapper JAR itself typically works across versions (in my experience) but I hear your point. There is also the checksum-validation issue... 

Maybe we could ask legal for clarification whether this particular JAR can be distributed as part of source releases? This restriction makes things annoyingly complex to maintain for close to zero benefit...

Reply:
https://issues.apache.org/jira/browse/LEGAL-288

Reply:
But then, see this:
https://github.com/search?q=org%3Aapache+filename%3Agradle-wrapper.jar

Maybe we're just trying to do too much... It really is making a simple and reliable thing a lot more complicated than it  needs to be.

Reply:
There is this discussion from 2014 https://lists.apache.org/thread.html/e19b21cadfa4efa67b8e9cf6f6425bc4c6714a947d11273c2fe8e9b2%401402431481%40%3Cgeneral.incubator.apache.org%3E

There is extant solution in Apache Groovy to use an already installed gradle to create the wrapper, but only for building src releases https://github.com/apache/groovy#bootstrapping-gradle

And it looks like there is no interest in allowing the wrapper jar in source releases via https://issues.apache.org/jira/browse/LEGAL-288

I'll cross-post this to the JIRA because I think this is important research to show.

Reply:
I see we found the same things here.

The other option is to exclude the gradle-wrapper from failing the smoke check? I don't want to take the position that because other projects are doing something which is against apache policy (and we even have a check for it!) that it's ok for us to do so as well.